hazelcast/src/main/java/com/hazelcast/cache/impl/DeferredValue.java

```diff
@@ -81,7 +85,12 @@ public Data getSerializedValue(SerializationService serializationService) {
             copy.serializedValueExists = true;
             copy.serializedValue = serializedValue;
         }
-        if (valueExists) {
+        if (!resolved && serializationService != null) {
```

Comment:
It's a bit hard reasoning about the states of `DeferredValue`. Isn't it the case that either `serializedValueExists` is `true` or `valueExists` is `true` and that we can simply use the previous code? Which scenario does this new case cover?

Reply:
This is a simple fix for NPE.

Reply:
When does NPE occur?

Reply:
Oh I really don't remember the details. There is a whole bunch of "other side" here in Payara that you don't see and it was quite the effort to get all of this working. I just fixed the NPE when I saw it and I unfortunately don't remember where exactly it was.